Course_Part_10/src/TicTacToeBasic.java

```diff
@@ -0,0 +1,247 @@
+     */
+    static void playerTurn() {
+        String playerMove;
+        do {
+            System.out.print("Please enter your move:");
+            playerMove = input.next();
+        } while (isMoveNotValid(playerMove));
+        playOptions.put(playerMove, "XX");
+    }
+
+    /**
+     * Using the computerPickMove method a random move will be picked and displayed
+     */
+    static void computerTurn() {
+        String computerMove = computerPickMove();
+        System.out.println("I will play at " + computerMove);
+        playOptions.put(computerMove, "OO");
+    }
+
+    /**
+     * Computer will pick a random move from the remaining options
+     *
+     * @return the random move
+     */
+    static String computerPickMove() {
+        ArrayList<String> remainingOptions = remainingPlayOptions();
+        Random randomIndex = new Random();
+        return remainingOptions.get(randomIndex.nextInt(remainingOptions.size()));
+    }
+
+    /**
+     * The board with all the options will be displayed at the start of the game
+     */
+    static void displayBoardStartGame() {
+        System.out.println("" +
+                " A1 | B1 | C1 \n" +
+                "--------------\n" +
+                " A2 | B2 | C2 \n" +
+                "--------------\n" +
+                " A3 | B3 | C3   "
+        );
+        System.out.println();
+    }
+
+    /**
+     * The board will be displayed according to the play mode chosen
+     * A: will show all the made moves and all the remaining options
+     * B: will only show all the made moves
+     */
+    static void displayBoard() {
+        String first;
+        String second;
+        String third;
+        System.out.println();
+        // Board will be printed layer by layer beginning from the top
+        for (int i = 1; i <= 3; i++) {
+            if (playMode.equals("A")) {
+                first = playOptions.get("A" + i).equals("  ") ? ("A" + i) : playOptions.get("A" + i);
+                second = playOptions.get("B" + i).equals("  ") ? ("B" + i) : playOptions.get("B" + i);
+                third = playOptions.get("C" + i).equals("  ") ? ("C" + i) : playOptions.get("C" + i);
+            } else {
+                first = playOptions.get("A" + i);
+                second = playOptions.get("B" + i);
+                third = playOptions.get("C" + i);
+            }
+
+            System.out.printf(" %s | %s | %s \n", first, second, third);
+            if (i != 3) {
+                System.out.println("--------------");
+            }
+        }
+        System.out.println();
+    }
+
+    /**
+     * Validate the player's input (move)
+     * A move is valid if it's one of the remaining options
+     *
+     * @param move is the player's input
+     * @return true if move is not valid and false is valid
+     */
+    static boolean isMoveNotValid(String move) {
+        ArrayList<String> remainingOptions = remainingPlayOptions();
```

Comment:
op verschillende plaatsen maak je een nieuwe ArrayList en initialiseert deze, wat ook kan is dat je naast de Map ook een ArrayList maakt die gevuld wordt met alle te spelen velden en elke keer dat een veld is gespeeld deze te verwijderen uit de lijst ..